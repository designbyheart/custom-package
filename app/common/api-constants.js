// @flow
// TODO:KS Consider moving it to api folder
export const TOKEN_EXPIRED_CODE = 'CRS-121'
export const SERVER_ERROR_CODE = 'OCS-000'
export const SERVER_API_CALL_ERROR = 'OCS-001'
export const PENDING_CONNECTION_REQUEST_CODE = 'CRS-103'
export const PUSH_NOTIFICATION_SENT_CODE = 'push-notification-sent'

export const QR_CODE_SIGNATURE = 's'
export const QR_CODE_CHALLENGE = 'c'
export const QR_CODE_REMOTE_CONNECTION_LOGO = 'lg'
export const QR_CODE_TARGET_DID = 'tDID'
export const QR_CODE_REMOTE_HOSTING_DID = 'rhDID'
export const QR_CODE_REMOTE_PAIR_WISE_DID = 'rpDID'
export const QR_CODE_TARGET_NAME = 'tn'
export const QR_CODE_SENDER_NAME = 'sn'
export const QR_CODE_UID = 'uid'

export const PUSH_NOTIFICATION_TYPE = {
  AUTH: 'auth-req',
  CLAIM_OFFER: 'claim-offer',
}
