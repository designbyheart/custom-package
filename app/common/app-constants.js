import { Platform } from 'react-native'

export const apptentiveCredentials = Platform.select({
  ios: {
    apptentiveKey: 'IOS-EVERNYM-ee7e2325084e',
    apptentiveSignature: '8049eba0f656a5b7aeb9722b84e5ec54',
  },
  android: {
    apptentiveKey: '<YOUR_ANDROID_APPTENTIVE_KEY>',
    apptentiveSignature: '<YOUR_ANDROID_APPTENTIVE_SIGNATURE>',
  },
})
