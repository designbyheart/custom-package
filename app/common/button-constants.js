export const ALLOW = 'Allow'
export const DENY = 'Deny'
