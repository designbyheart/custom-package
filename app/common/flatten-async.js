// @flow

// TODO:KS
// Explain the need for this function

export const flattenAsync = (
  fn: (...variableArgs: any[]) => Promise<any>
) => async (
  ...args: Array<any>
): Promise<[null | typeof Error, null | any]> => {
  try {
    const response = await fn(...args)
    return [null, response]
  } catch (e) {
    return [e, null]
  }
}
