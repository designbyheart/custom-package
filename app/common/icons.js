// @flow
import React from 'react'
import { Icon } from 'react-native-eva-icons'
import { colors } from './styles/constant'
import { moderateScale } from 'react-native-size-matters'

type IconProps = {
  width?: number,
  height?: number,
  color?: string,
  style?: any,
}

type CommonIconProps = {
  name: string,
} & IconProps

export const HOME_ICON = 'home-outline'
export const CONNECTIONS_ICON = 'people-outline'
export const CREDENTIALS_ICON = 'home-outline'
export const SETTINGS_ICON = 'settings-2-outline'

export const CAMERA_ICON = 'camera-outline'
export const CHECK_MARK_ICON = 'checkmark-circle-2-outline'
export const HOME_MENU_ICON = 'menu-outline'
export const ANDROID_BACK_ARROW_ICON = 'arrow-back-outline'
export const IOS_BACK_ARROW_ICON = 'arrow-ios-back-outline'
export const MORE_ICON = 'more-vertical-outline'
export const DELETE_ICON = 'trash-2-outline'
export const CLOSE_ICON = 'close-outline'

// common icon class is implemented to set default values (except name) for icons
// in that case we should only set icon name and get an icon with default color and size
export const EvaIcon = (props: CommonIconProps) => {
  const { name, width, height, color, style } = props

  return (
    <Icon
      name={name}
      width={width ? width : moderateScale(22)}
      height={height ? height : moderateScale(22)}
      fill={color ? color : colors.cmGray2}
      style={style}
    />
  )
}
