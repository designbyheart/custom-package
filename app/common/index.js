// @flow
export * from './api-constants'
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
