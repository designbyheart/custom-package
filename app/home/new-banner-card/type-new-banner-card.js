// @flow
export type NewBannerCardProps = {
  navigation: Object,
  navigationRoute: string,
  timestamp: string,
  logoUrl: string,
  uid: string,
  issuerName: string,
}
