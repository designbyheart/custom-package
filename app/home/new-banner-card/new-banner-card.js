// @flow
import React from 'react'
import { TouchableOpacity, Text, View, Image, StyleSheet } from 'react-native'
import { isiPhone5 } from '../../common/styles'
import { colors, font, fontFamily, fontSizes } from '../../common/styles/constant'
import { scale, verticalScale, moderateScale } from 'react-native-size-matters'

import type { NewBannerCardProps } from './type-new-banner-card'
import { DefaultLogo } from '../../components/default-logo/default-logo'

export const NewBannerCard = (props: NewBannerCardProps) => {
  return (
    <TouchableOpacity
      style={styles.container}
      onPress={() =>
        props.navigation.navigate(props.navigationRoute, { uid: props.uid })
      }
    >
      <View style={styles.iconSection}>
        {typeof props.logoUrl === 'string' ? (
          <Image source={{ uri: props.logoUrl }} style={styles.issuerLogo} />
        ) : (
          <DefaultLogo
            text={props.issuerName[0]}
            size={moderateScale(34, 0.15)}
            fontSize={isiPhone5 ? font.size.M : font.size.ML}
          />
        )}
      </View>
      <View style={styles.textSection}>
        <View style={styles.textIssuerSection}>
          <Text
            style={styles.issuerText}
            ellipsizeMode="tail"
            numberOfLines={1}
          >
            {props.issuerName}
          </Text>
        </View>
        <View style={styles.textMessageSection}>
          <Text style={styles.newMessageText}>NEW MESSAGE - TAP TO OPEN</Text>
        </View>
      </View>
      <View style={styles.textDateSection}>
        <Text style={styles.dateText}>{props.timestamp}</Text>
      </View>
    </TouchableOpacity>
  )
}

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    backgroundColor: colors.cmGreen3,
    borderWidth: 1,
    borderColor: colors.cmGreen1,
    height: moderateScale(70, 0.12),
    marginLeft: moderateScale(7, 0.1),
    marginRight: moderateScale(7, 0.1),
    marginTop: moderateScale(7, 0.1),
    borderRadius: 8,
  },
  iconSection: {
    height: '100%',
    width: moderateScale(65, 0.2),
    alignItems: 'center',
    justifyContent: 'center',
  },
  textSection: {
    flex: 1,
  },
  textIssuerSection: {
    flex: 2,
    justifyContent: 'flex-end',
  },
  textMessageSection: {
    flex: 2,
    justifyContent: 'flex-start',
  },
  textDateSection: {
    height: '100%',
    width: moderateScale(65, 0.2),
    alignItems: 'flex-end',
    justifyContent: 'flex-start',
  },
  issuerText: {
    fontFamily: fontFamily,
    fontSize: moderateScale(fontSizes.size5, 0.1),
    fontWeight: 'bold',
    color: colors.cmGray1,
    marginBottom: moderateScale(3, 0.1),
  },
  newMessageText: {
    fontFamily: fontFamily,
    fontSize: moderateScale(fontSizes.size8, 0.1),
    fontWeight: 'bold',
    color: colors.cmGray1,
    marginTop: moderateScale(3, 0.1),
  },
  issuerLogo: {
    width: moderateScale(34, 0.15),
    height: moderateScale(34, 0.15),
    borderRadius: moderateScale(34, 0.15) / 2,
  },
  dateText: {
    fontFamily: fontFamily,
    fontWeight: '500',
    fontSize: moderateScale(fontSizes.size9, 0.1),
    fontStyle: 'italic',
    color: colors.cmGray1,
    marginTop: moderateScale(8, 0.1),
    marginRight: moderateScale(8, 0.1),
  },
  placeholderIfNoImage: {
    width: moderateScale(34, 0.15),
    height: moderateScale(34, 0.15),
    borderRadius: moderateScale(34, 0.15) / 2,
    backgroundColor: colors.cmGray2,
    alignItems: 'center',
    justifyContent: 'center',
  },
  placeholderTextIfNoImage: {
    fontFamily: fontFamily,
    fontSize: moderateScale(fontSizes.size4, 0.1),
    fontWeight: 'bold',
    color: colors.cmWhite,
  },
})
