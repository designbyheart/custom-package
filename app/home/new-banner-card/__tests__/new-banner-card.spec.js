// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import 'react-native'

import { getNavigation } from '../../../../__mocks__/static-data'
import { NewBannerCard } from '../new-banner-card'
import { claimOfferRoute } from '../../../common'

describe('<NewBannerCard />', () => {
  const navigation = getNavigation()

  it('should render NewBannerCard component and match snapshot', () => {
    const component = renderer
      .create(
        <NewBannerCard
          navigation={navigation}
          navigationRoute={claimOfferRoute}
          timestamp="2020-03-15T15:57:54+01:00"
          logoUrl="https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSjimxZbcvZzPcPvHd_y7f0tc5d8QoC9DOPecb8JTOChmS1IoDq"
          uid="MGEwNzg"
          issuerName="Evernym QA-RC"
        />
      )
      .toJSON()
    expect(component).toMatchSnapshot()
  })
})
