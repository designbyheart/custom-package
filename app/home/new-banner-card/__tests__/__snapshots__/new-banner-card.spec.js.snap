// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewBannerCard /> should render NewBannerCard component and match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "rgba(134, 185, 59, 0.15)",
      "borderColor": "#86B93B",
      "borderRadius": 8,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 69.28,
      "marginLeft": 6.9399999999999995,
      "marginRight": 6.9399999999999995,
      "marginTop": 6.9399999999999995,
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "100%",
        "justifyContent": "center",
        "width": 63.885714285714286,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcSjimxZbcvZzPcPvHd_y7f0tc5d8QoC9DOPecb8JTOChmS1IoDq",
        }
      }
      style={
        Object {
          "borderRadius": 16.78142857142857,
          "height": 33.56285714285714,
          "width": 33.56285714285714,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
          "justifyContent": "flex-end",
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#505050",
            "fontFamily": "Lato",
            "fontSize": 14.871428571428572,
            "fontWeight": "bold",
            "marginBottom": 2.974285714285714,
          }
        }
      >
        Evernym QA-RC
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 2,
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#505050",
            "fontFamily": "Lato",
            "fontSize": 10.905714285714286,
            "fontWeight": "bold",
            "marginTop": 2.974285714285714,
          }
        }
      >
        NEW MESSAGE - TAP TO OPEN
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "height": "100%",
        "justifyContent": "flex-start",
        "width": 63.885714285714286,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#505050",
          "fontFamily": "Lato",
          "fontSize": 9.914285714285715,
          "fontStyle": "italic",
          "fontWeight": "500",
          "marginRight": 7.9314285714285715,
          "marginTop": 7.9314285714285715,
        }
      }
    >
      2020-03-15T15:57:54+01:00
    </Text>
  </View>
</View>
`;
